modules/baseline/src/main/java/org/apache/ignite/internal/baseline/BaselineManager.java

```diff
@@ -31,32 +38,29 @@
 // TODO: https://issues.apache.org/jira/browse/IGNITE-14716 Adapt concept of baseline topology IEP-4.
 @SuppressWarnings({"FieldCanBeLocal", "unused"})
 public class BaselineManager implements IgniteComponent {
-    /** Configuration manager in order to handle and listen baseline specific configuration. */
-    private final ConfigurationManager configurationMgr;
-
-    /**
-     * MetaStorage manager in order to watch private distributed baseline specific configuration, cause ConfigurationManger handles only
-     * public configuration.
-     */
-    private final MetaStorageManager metastorageMgr;
+    /** Cluster configuration in order to handle and listen baseline specific configuration. */
+    private final ClusterConfiguration clusterConfiguration;
 
     /** Cluster network service in order to retrieve information about current cluster nodes. */
     private final ClusterService clusterSvc;
 
+    /** Busy lock to stop synchronously. */
+    private final IgniteSpinBusyLock busyLock = new IgniteSpinBusyLock();
+
+    /** Prevents double stopping the component. */
+    AtomicBoolean stopGuard = new AtomicBoolean();
```

Comment:
IMHO, I don't think, that this flag is needed.

Reply:
I dont think, that is a good idea to tune concrete implementation for test purposes. In my understanding, this should be properly handled by test itself. Perhaps, we need to introduce something like `IgniteComponentAdapter` that should take care of starting/stopping a component multiple times.